pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -841,6 +861,73 @@ func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMe
 	return &machineapierrors.RequeueAfterError{RequeueAfter: time.Second * 5}
 }
 
+// isActionTimedOut checks if time since timestamp stored in specified machine annotation did not exceed given timeout
+func isActionTimedOut(machine *machinev1beta1.Machine, annotation string) bool {
```

Comment:
It seems kind of weird that we're passing the annotation name as a parameter here.

Reply:
I agree. Is that just so the unit tests can use a different name?

Reply:
It is a left-over of my original plan to also handle power off action. It is not needed anymore.